libexec/tgenv-upgrade

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -e
+
+[ -n "${TGENV_DEBUG}" ] && set -x
+source "${TGENV_ROOT}/libexec/helpers"
+
+echo "Upgrading tgenv to the latest version..."
+
+if [ -d "$HOME/.tgenv" ]; then
+  cd "$HOME/.tgenv"
```

Comment:
I would recommend not hardcoding `$HOME/.tgenv` here

`${TGENV_ROOT}` tells us where we need to CD to to do the upgrade. It probably points to `$HOME/.tgenv` but  in case it doesn't, then.

Using `${TGENV_ROOT}` also saves us from having to even check if `$HOME/.tgenv` exists  becuase we can presume that since we are running this script, it exists.

It could be wise to check that TGENV is set though so we dont accidentally end up doing  `cd ""`

```
if [ -n "${TGENV_ROOT}" ]; then
  cd "${TGENV_ROOT}"
  # ...
fi
```

Reply:
I've changed to `${TGENV_ROOT}` and try include a check for verify the `export` of it.